src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -855,7 +855,7 @@ protected Object deserializeUsingPropertyBasedWithUnwrapped(JsonParser p, Deseri
             if (buffer.readIdProperty(propName) && creatorProp == null) {
                 continue;
             }
-            if (creatorProp != null) {
+            if (creatorProp != null && !_unwrappedPropertyHandler.isUnwrapped(creatorProp)) {
```

Comment:
I'll have to think about this; ideally such lookup shouldn't be needed (rather, creatorProp would have knowledge of whether it's created via unwrapping).

Reply:
Oddly enough, removing this call makes no difference; all tests pass the same (including Record test).

So is it actually needed? Either we should have a test that fails without it, or it should just be removed, I think.

Reply:
Yeah, it seems like this makes no sense anymore, I probably implemented it before adapting the rest of the code. The issue this _would_ have solved is if you have a structure like this:
```java
record Outer(@JsonUnwrapped Inner propertyName) {}

record Inner(String propertyName, String otherPropertyName) {}
```
There is a property called `propertyName` inside and outside the unwrapped structure, so when deserializing and encountering the property called "propertyName", I initially thought it would find that property. But as far as this code is concerned, the outer property is called `"@JsonUnwrapped"`, so it should not clash either way. Because of this fake name, the `_unwrappedPropertyNames` set used for `isUnwrapped` actually only contains `"@JsonUnwrapped"`... 😄

Reply:
Right, I thought it might just contain `@JsonUnwrapped` :)